packages/vda-node-manager/src/blockchain/ownerApi.ts

```diff
@@ -0,0 +1,361 @@
+        if (this.readOnly || !this.config.signKey) {
+            throw new Error(`Unable to submit to blockchain. In read only mode.`)
+        }
+
+        const response = await this.vdaWeb3Client!.setStakingRequired(isRequired);
+        
+        if (response.success !== true) {
+            throw new Error(`Failed to set staking required: ${response.reason}`);
+        }
+
+    }
+
+    /**
+     * Update the `STAKE_PER_SLOT` value of contract
+     * @param stakeAmount The new value to be updated
+     */
+    public async updateStakePerSlot(stakeAmount: BigNumberish) {
+        if (this.readOnly || !this.config.signKey) {
+            throw new Error(`Unable to submit to blockchain. In read only mode.`)
+        }
+
+        const response = await this.vdaWeb3Client!.updateStakePerSlot(stakeAmount);
+        
+        if (response.success !== true) {
+            throw new Error(`Failed to set StakePerSlot: ${response.reason}`);
+        }
+    }
+
+    /**
+     * Update the minmum of `STAKE_PER_SLOT` value
+     * @param minSlots The new value to be updated
+     */
+    public async updateMinSlotCount(minSlots: BigNumberish) {
+        if (this.readOnly || !this.config.signKey) {
+            throw new Error(`Unable to submit to blockchain. In read only mode.`)
+        }
+
+        const response = await this.vdaWeb3Client!.updateMinSlotCount(minSlots);
+        
+        if (response.success !== true) {
+            throw new Error(`Failed to update minimum slot count: ${response.reason}`);
+        }
+    }
+
+    /**
+     * Update the maximum of `STAKE_PER_SLOT` value
+     * @param maxSlots The new value to be updated
+     */
+    public async updateMaxSlotCount(maxSlots: BigNumberish) {
+        if (this.readOnly || !this.config.signKey) {
+            throw new Error(`Unable to submit to blockchain. In read only mode.`)
+        }
+
+        const response = await this.vdaWeb3Client!.updateMaxSlotCount(maxSlots);
+        
+        if (response.success !== true) {
+            throw new Error(`Failed to update maximum slot count: ${response.reason}`);
+        }
+    }
+    
+    /**
+     * Update the fee for logging a node issue
+     * @param newFee New fee value to be set.
+     */
+    public async updateNodeIssueFee(newFee: BigNumberish) {
+        if (this.readOnly || !this.config.signKey) {
+            throw new Error(`Unable to submit to blockchain. In read only mode.`)
+        }
+
+        const response = await this.vdaWeb3Client!.updateNodeIssueFee(newFee);
+        
+        if (response.success !== true) {
+            throw new Error(`Failed to update fee to log a node issue : ${response.reason}`);
+        }
+    }
+
+    /**
+     * Withdraw the VDA tokens that was deposited by `logNodeIssue()` function
+     * @param to Receiving address
+     * @param amount Amount to be withdrawn
+     */
+    public async withdrawIssueFee(to: string, amount: BigNumberish) {
```

Comment:
Is this required? I thought we just have `withdraw()` now?

Reply:
This function is not the same as `withdraw()` function.
The `withdraw()` function in the `vda-node-manger/src/blockchain/userapi.ts` is for users to withdraw their staked tokens.

The `withdrawIssueFee()` function in the `vda-node-manager/src/blockchain/ownerApi.ts` is for owner to withdraw the fees from the `logNodeIssue()` function.

Reply:
Ok, got it, thanks.